File: SymbolTable.swift
/* SYMBOL TABLE */

class SymbolTable {
    var symbols = [String:Number]()
    var parent:SymbolTable? = nil 

    init() {
        self.symbols = [String : Number]()
        self.parent = nil
    }

    func get_val(name: String) -> Number {
        let value = symbols[name]
        var returnVal: Number = Number(0.0)

        if let v = value {
            returnVal = v 
        }

        if let p = parent {
            returnVal = p.get_val(name: name)
        }

        return returnVal
    }

    func set_val(name: String, value: Number) {
        self.symbols[name] = value 
    }

    func remove_val(name: String) {
        self.symbols.removeValue(forKey: name)
    }
}

File: src/SymbolTable.py
class SymbolTable:
    def __init__(self, symbols={}, parent=None):
        self.symbols = symbols
        self.parent = parent

    def get_val(self, name):
        from Types import Number

        value = self.symbols[name]
        return_val = Number(0)

        if value is not None:
            return_val = value

        if self.parent is not None:
            return_val = self.parent.get_val(name)

        return return_val

    def set_val(self, name, value):
        self.symbols[name] = value

    def remove_val(self, name):
        del self.symbols[name]
